src/families/cosmos/bridge/libcore.js

```diff
@@ -0,0 +1,237 @@
+// @flow
+import { scanAccounts } from "../../../libcore/scanAccounts";
+import { sync } from "../../../libcore/syncAccount";
+import type { AccountBridge, CurrencyBridge } from "../../../types";
+import type { Transaction } from "../types";
+import { BigNumber } from "bignumber.js";
+import broadcast from "../libcore-broadcast";
+import signOperation from "../libcore-signOperation";
+import { getMainAccount } from "../../../account";
+import { makeLRUCache } from "../../../cache";
+import { validateRecipient } from "../../../bridge/shared";
+import { getFeesForTransaction } from "../../../libcore/getFeesForTransaction";
+import {
+  NotEnoughBalance,
+  InvalidAddressBecauseDestinationIsAlsoSource,
+  InvalidAddress,
+} from "@ledgerhq/errors";
+import {
+  CosmosTooMuchRedelegations,
+  CosmosTooMuchUnboundings,
+  CosmosRedelegationInProgress,
+} from "../../../errors";
+import {
+  setCosmosPreloadData,
+  asSafeCosmosPreloadData,
+} from "../preloadedData";
+import { getValidators, hydrateValidators } from "../validators";
+
+const calculateFees = makeLRUCache(
+  async (a, t) => {
+    return getFeesForTransaction({
+      account: a,
+      transaction: t,
+    });
+  },
+  (a, t) =>
+    `${a.id}_${t.amount.toString()}_${t.recipient}_${
+      t.gasLimit ? t.gasLimit.toString() : ""
+    }_${t.fees ? t.fees.toString() : ""}
+    _${String(t.useAllAmount)}_${t.mode}_${
+      t.validators ? t.validators.map((v) => v.address).join("-") : ""
```

Comment:
doesn't `memo` should make the fee recalculate?
same question for cosmosSourceValidator

Reply:
technically it should, but hopefully this is covered by the multiplier on the initial gas estimate. A "send" typically uses 30/40k Gas, and memo costs 10 gas per byte

the cosmosSourceValidator might have effects I don't know of, but theoretically this shouldn't change the cost much